mythril_core/src/acpi.rs

```diff
@@ -0,0 +1,84 @@
+#![deny(missing_docs)]
+use core::fmt;
+
+/// Extented BIOS Data Area Start Address
+const EXTENDED_BIOS_DATA_START: usize = 0x0000040e;
+/// Extented BIOS Data Area End Address
+const EXTENDED_BIOS_DATA_END: usize = 0x0000040e + 2048;
+/// Main BIOS Data Area Start Address
+const MAIN_BIOS_DATA_START: usize = 0x000e0000;
+/// Main BIOS Data Area End Address
+const MAIN_BIOS_DATA_END: usize = 0x000fffff;
+/// Well Known RSDP Signature
+const RSDP_SIGNATURE: &[u8; 8] = b"RSD PTR ";
+
+/// The Root System Descriptor Pointer (RSDP)
+#[derive(Copy, Clone, Debug)]
+#[repr(packed)]
+pub struct RSDP {
+    signature: [u8; 8],
+    checksum: u8,
+    oemid: [u8; 6],
+    revision: u8,
+    rsdt_addr: u32,
+    length: u32,
+    xsdt_addr: u64,
+    ext_checksum: u8,
+    reserved: [u8; 3]
+}
+
+impl fmt::Display for RSDP {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        unsafe {
```

Comment:
why does this need unsafe?

Reply:
technically borrowing from a packed struct... not _technically_ an error yet, but it will be soon

Reply:
You're referring to https://github.com/rust-lang/rust/issues/46043? That's a fair point.